Demos/Aspose.Email.Cloud.Live.Demos/Aspose.Email.Cloud.Live.Demos.UI/Scripts/Common/filedrop.js

```diff
@@ -0,0 +1,268 @@
+﻿// File Upload Component
+// options:
+// - DropFilesPrompt (string)
+// - ChooseFilePrompt (string)
+// - Accept (string)
+// - Multiple (boolean)
+// - UseSorting (boolean)
+// - MaximumUploadFiles (number)
+// - FileWrongTypeMessage (string)
+// - FileAmountMessage (string)
+// - FileSelectMessage (string)
+// - UploadOptions (array of string)
+// - UIBasePath (string)
+// - Method (callback)
+// - UploadAndRedirect (boolean)
+// - showAlert (function (string) => void)
+// - hideAlert (function () => void)
+// - showLoader (function () => void)
+// - progress (function () => void)
+
+(function ($) {
+    $.fn.filedrop = function (options) {
+        var getRandomIntInclusive = function (min, max) {
+            min = Math.ceil(min);
+            max = Math.floor(max);
+            return Math.floor(Math.random() * (max - min + 1)) + min;
+        };
+
+        var randomId = getRandomIntInclusive(1, Number.MAX_SAFE_INTEGER);
+        var acceptExts = options.Accept.split(/\s*,\s*/).map(function (ext) {
+            return ext.substring(1).toUpperCase();
+        });
+
+        var droppedFiles = [];
+
+        var getFileExtension = function (file) {
+            var pos = file.name.lastIndexOf('.');
+            return pos !== -1 ? file.name.substring(pos + 1).toUpperCase() : null;
+        };
+
+        var nextFileId = function () {
+            var id = 1;
+            var found;
+            do {
+                found = false;
+
+                for (let i = 0; i < droppedFiles.length; i++) {
+                    if (droppedFiles[i].id === id) {
+                        id += 1;
+                        found = true;
+                        break;
+                    }
+                }
+            } while (found);
+            return id;
+        };
+
+        var preventFileDrop = function (evt) {
+            evt = evt || event;
+            evt.preventDefault();
+            evt.stopPropagation();
+        };
+
+        var removeFileBlock = function (id) {
+            var pos;
+            for (pos = 0; pos < droppedFiles.length; pos++) {
+                if (droppedFiles[pos].id === id) {
+                    break;
+                }
+            }
+            if (pos < droppedFiles.length) {
+                droppedFiles.splice(pos, 1);
+                $('#filedrop-' + randomId).find('#fileupload-' + id).remove();
+                if (droppedFiles.length === 0) {
+                    $('#filedrop-' + randomId).find('.chooseFilesLabel').removeClass('hidden');
+                }
+            }
+        };
+
+        var moveUpFileBlock = function (id) {
+            var pos;
+            for (pos = 0; pos < droppedFiles.length; pos++) {
+                if (droppedFiles[pos].id === id) {
+                    break;
+                }
+            }
+            if (pos < droppedFiles.length && pos !== 0) {
+                var prevId = droppedFiles[pos - 1].id;
```

Comment:
You should create a function `swapFiles` and call it for both `moveUp` and `moveDown`:
```
swapFiles(pos, pos - 1); //for move up
swapFiles(pos, pos + 1); //for move down
```